operations/security_and_authentication/Configure_root_user_on_HPE_iLO_BMCs.md

```diff
@@ -0,0 +1,261 @@
+
+    ```bash
+    curl -k -u "$DEFAULT_USERNAME:$DEFAULT_PASSWORD" https://$BMC/redfish/v1/Managers -i | head -1
+    ```
+
+    If a `200 OK` status code is returned, then the default user account is configured correctly.
+
+    ```text
+    HTTP/1.1 200 OK
+    ```
+
+    If a `401 Unauthorized` status code is returned, the the default user is not configured correctly. The BMC needs to be factory reset to restore the default user credentials.
+
+    ```text
+    HTTP/1.1 401 Unauthorized
+    ```
+
+1. (`ncn#`) **Do only if a `401 Unauthorized` was observed in the previous step**, factory reset BMCs back to factory defaults:
+
+    1. Follow [this HPE support article](https://techlibrary.hpe.com/docs/iss/proliant-gen10-uefi/s_reset_ilo_defaults.html) to boot the node into its BIOS and perform a factory reset on the BMC.
+        > **NOTE**: If this is a Apollo 6500 XL 645D Gen10 Plus node the BMC will need to be reconfigured with the expected network settings.
+        > See [Configure HPE Apollo 6500 XL645d Gen10 Plus Compute Nodes](../../install/prepare_compute_nodes.md#configure-hpe-apollo-6500-xl645d-gen10-plus-compute-nodes) for more information.
+
+    1. Verify the the BMC can be accessed with the default credentials after the factory reset has been performed.
+
+        ```bash
+        curl -k -u "$DEFAULT_USERNAME:$DEFAULT_PASSWORD" https://$BMC/redfish/v1/Managers -i  | head -1
+        ```
+
+        Expected output:
+
+        ```text
+        HTTP/1.1 200 OK
+        ```
+
+1. (`ncn#`) Configure the root user account:
+
+    1. Determine if the root user account already exists:
+
+        ```bash
+        for account in $(curl -s -u $DEFAULT_USERNAME:$DEFAULT_PASSWORD -k https://$BMC/redfish/v1/AccountService/Accounts | jq '.Members[]."@odata.id"' -r); do 
+            echo "Checking $account"; 
+
+            curl -k -s -u $DEFAULT_USERNAME:$DEFAULT_PASSWORD -k https://${BMC}${account} | jq '. | {Id: .Id, UserName: .UserName, RoleId: .RoleId}' -c
+        done
+        ```
+
+        Example output if the root user exists:
+
+        ```json
+        Checking /redfish/v1/AccountService/Accounts/1
+        {"Id":"1","UserName":"Administrator","RoleId":"Administrator"}
+        Checking /redfish/v1/AccountService/Accounts/2
+        {"Id":"2","UserName":"root","RoleId":"Administrator"}
+        ```
+
+        Example output if the root user does not exist:
+
+        ```json
+        Checking /redfish/v1/AccountService/Accounts/1
+        {"Id":"1","UserName":"Administrator","RoleId":"Administrator"}
+        ```
+
+    1. **If the user does not exist**, then create the root user account:
+
+        ```bash
+        curl -k -u "$DEFAULT_USERNAME:$DEFAULT_PASSWORD" -X POST \
+                -H 'Content-Type: application/json' \
+                -d $(jq --arg PASSWORD "$EXPECTED_ROOT_PASSWORD" -nc '{RoleId: "Administrator", UserName: "root", Password: $PASSWORD}') \
```

Comment:
Still not good to pass passwords on the command line, but I had never considered using jq this way for creating payloads for curl. Very slick.

Reply:
Might be able to make the JQ access the password from the environment. https://stedolan.github.io/jq/manual/#$ENV,env

Also in the future I hope this entirely procedure here would be superseded by FR5 in https://github.com/Cray-HPE/community/issues/37